src/ethereum_spec_tools/diff.py

```diff
@@ -40,30 +40,23 @@ def find_pickles(path: str, fork: Hardfork) -> Iterator[str]:
 
 
 def meaningful_diffs(
-    pub: rstdiff.Publisher3Args,
     old: Any,
     new: Any,
+    trivial_changes: List[Tuple[str, str, re.RegexFlag]],
 ) -> bool:
     """Find if there are meaningful differences between the docs"""
-    realDebug = pub.settings.debug
-    pub.settings.debug = pub.settings.dump_rstdiff
-    reporter = new_reporter("RSTDIFF", pub.settings)
-    pub.settings.debug = realDebug
-    dispatcher = rstdiff.DocutilsDispatcher(reporter)
-    opcodes = rstdiff.doDiff(dispatcher, old, new)
-
-    opcode_counter = rstdiff.OpcodeCounter(opcodes)
-    opcode_counter.count()
-
-    return bool(
-        opcode_counter.replace
-        or opcode_counter.delete
-        or opcode_counter.insert
-    )
+    old = re.sub(r"[\n\t\s]", "", str(old))
+    new = re.sub(r"[\n\t\s]", "", str(new))
+
+    for old_word, new_word, flag in trivial_changes:
+        old = re.sub(old_word, "", old, flags=flag)
+        new = re.sub(new_word, "", new, flags=flag)
```

Comment:
`old_word` and `new_word` are escaped?

I assume we're using `re.sub` instead of `str.replace` for the case-insensitivity?

Reply:
For case insensitivity as well as to identify other corner cases in class names and other places. For example - `dao fork` becomes `dao-fork`, `dao_fork` etc.